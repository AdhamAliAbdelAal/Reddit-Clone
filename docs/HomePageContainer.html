<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: HomePageContainer</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: HomePageContainer</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>HomePageContainer<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="HomePageContainer"><span class="type-signature"></span>new HomePageContainer<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h4>
    

    



<div class="description">
    <p>This component works as a container for all home page components</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="HomePage_HomePageContainer_HomePageContainer.jsx.html">HomePage/HomePageContainer/HomePageContainer.jsx</a>, <a href="HomePage_HomePageContainer_HomePageContainer.jsx.html#line22">line 22</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Container represents the home page</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">React.Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="CreatePostInHome.html">CreatePostInHome</a></li><li><a href="HomePageContainer.html">HomePageContainer</a></li><li><a href="HomePageFooter.html">HomePageFooter</a></li><li><a href="JoinButton.html">JoinButton</a></li><li><a href="MainContent.html">MainContent</a></li><li><a href="PersonalReddit.html">PersonalReddit</a></li><li><a href="Post.html">Post</a></li><li><a href="PostHeader.html">PostHeader</a></li><li><a href="PostList.html">PostList</a></li><li><a href="PostReactions.html">PostReactions</a></li><li><a href="PostsClassification.html">PostsClassification</a></li><li><a href="Reactions.html">Reactions</a></li><li><a href="RedditPremium.html">RedditPremium</a></li><li><a href="SideBar.html">SideBar</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.0</a> on Mon Nov 07 2022 22:11:59 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>