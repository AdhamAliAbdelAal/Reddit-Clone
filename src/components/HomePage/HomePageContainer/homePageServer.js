import { redirectLogin } from '../../../utils/Redirect';
import useFetch from '../../../hooks/useFetch';
import useFetchParams from '../../../hooks/useFetchParams';

const homePageServer = (postClass) => {
  const postsUrl = `/users/${postClass || 'best'}`;
  // eslint-disable-next-line no-unused-vars
  const [data, postsError, statusCode] = useFetchParams(postsUrl);
  const dbPosts = data?.data;
  // console.log('from home', statusCode);
  console.log('posts', dbPosts?.length);
  const posts = dbPosts?.map((post) => {
    // for backend bugs
    const temp = {
      _id: 1,
      text: '',
      backgroundColor: '',
      textColor: '',
    };
    const {
      text: flairText,
      backgroundColor: flairBackgroundColor,
      textColor: flairColor,
    } = (post.flairId) || temp;
    const { name: ownerName, icon: ownerIcon } = post.owner;
    const { name: authorName } = post.author;
    return ({
      ...post,
      flairText,
      flairBackgroundColor,
      flairColor,
      ownerName,
      ownerIcon,
      authorName,
    });
  });
  // console.log('posts final', posts, posts?.length);

  const [communitiesData, communitiesError, communitiesStatusCode] = useFetch('/subreddits/random/leaderboard?limit=4&page=0');
  if (communitiesStatusCode === 401) {
    redirectLogin();
  }
  return [posts, postsError, communitiesData?.data, communitiesError];
};

export default homePageServer;
