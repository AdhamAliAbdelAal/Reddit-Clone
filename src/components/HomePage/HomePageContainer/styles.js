import { Box } from '@mui/material';
import { styled } from '@mui/system';

export const MainContainer = styled('div')({
  display: 'flex',
  justifyContent: 'center',
});

export const OuterContainer = styled(Box)({
  backgroundColor: '#DAE0E6',
  width: '100%',
  height: '250vh',
});
