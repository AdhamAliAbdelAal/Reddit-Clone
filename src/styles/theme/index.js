import { createTheme } from '@mui/material';

const theme = createTheme({
  palette: {
    primary: {
      main: '#0079D3',
      light: '#1484D6',
      dark: '#1484D6',
      fade: '#3394DC',
    },
    secondary: {
      main: '#FF4500',
      light: '#FF5414',
      dark: '#FF5414',
    },
    third: {
      main: '#898989',
      light: '#E8E8E8',
      dark: '#E8E8E8',
    },
    action: {
      hoverOpacity: 0.1,
    },
    neutral: {
      main: '#978A8C',
      contrastText: '#fff',
    },
    white: {
      main: '#ffffff',
      light: '#ffffff',
      dark: '#898989',
    },
    offwhite: {
      main: '#DAE0E6',
      dark: '#2E2F30',
    },
  },
});

export const fonts = {
  'system-ui': 'system-ui',
};
export default theme;
