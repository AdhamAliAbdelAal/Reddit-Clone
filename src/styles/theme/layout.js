import { createTheme } from '@mui/material';

const layoutTheme = (theme) => createTheme({
  ...theme,
  breakpoints: {
    values: {
      xxs: 335,
      xs: 420,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
});
export default layoutTheme;
