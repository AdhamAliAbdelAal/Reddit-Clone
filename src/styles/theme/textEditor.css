.rdw-option-wrapper {
    background-color: #f6f7f8 !important;
    padding: 5px 3px !important;
    border-radius: 7px !important;
    border-width: 0 !important;
    margin: 0 3px !important;
  }
  .rdw-option-wrapper:hover,.rdw-option-active{
    background-color: #1a1a1b1a !important;
    box-shadow: none !important;
  }

.public-DraftStyleDefault-block{
    margin: 0 !important;
    margin-top: 5px !important;
}