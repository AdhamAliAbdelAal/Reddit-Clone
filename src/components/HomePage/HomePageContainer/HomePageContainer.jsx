import { useMediaQuery, useTheme } from '@mui/material';
import { useParams } from 'react-router-dom';
import MainContent from '../../MainContent/MainContent';
import { MainContainer, OuterContainer } from './styles';
import SideBar from '../../SideBar/SideBar';
import Communities from '../Communities/Communities';
import RedditPremium from '../RedditPremium/RedditPremium';
import PersonalReddit from '../PersonalReddit/PersonalReddit';
import HomePageFooter from '../HomePageFooter/HomePageFooter';
import PostsClassification from '../PostsClassification/PostsClassification';
import CreatePostInHome from '../CreatePostInHome/CreatePostInHome';
import BackToTop from '../../BackToTop/BackToTop';
import useFetch from '../../../hooks/useFetch';
import PostList from '../PostList/PostList';
/**
 * This component works as a container for all home page components
 * and a repository the data fetched in
 *
 * @component HomePageContainer
 * @returns {React.Component} Container represents the home page
 */

function HomePageContainer() {
  const { postClass } = useParams();
  // variables
  const theme = useTheme();
  const match = useMediaQuery(theme.breakpoints.up('md'));
  const communitiesUrl = 'subreddits/mine/subscriber';
  console.log(postClass);
  const postsUrl = `/users/${postClass || 'best'}`;
  // console.log(postsUrl);

  // states
  const [posts, postsError] = useFetch(postsUrl);
  const [communities, communitiesError] = useFetch(communitiesUrl);

  return (
    <OuterContainer>
      <MainContainer>
        <MainContent width={640}>
          <CreatePostInHome />
          <PostsClassification />
          {!postsError ? (posts && <PostList posts={posts} />) : 'error in fetching posts'}
        </MainContent>
        {match
        && (
        <SideBar>
          {!communitiesError ? (communities && <Communities communities={communities} />) : 'error in fetching communities'}
          <RedditPremium />
          <PersonalReddit />
          <HomePageFooter />
          <BackToTop />
        </SideBar>
        )}
      </MainContainer>
    </OuterContainer>
  );
}

export default HomePageContainer;
