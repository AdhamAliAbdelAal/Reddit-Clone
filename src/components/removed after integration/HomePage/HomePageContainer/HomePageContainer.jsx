import { useMediaQuery, useTheme } from '@mui/material';
import MainContent from '../../MainContent/MainContent';
import { MainContainer, OuterContainer } from './styles';
import SideBar from '../../SideBar/SideBar';
import Post from '../../Post/Post';
import Communities from '../Communities/Communities';
import RedditPremium from '../RedditPremium/RedditPremium';
import PersonalReddit from '../PersonalReddit/PersonalReddit';
import HomePageFooter from '../HomePageFooter/HomePageFooter';
import PostsClassification from '../PostsClassification/PostsClassification';
import CreatePostInHome from '../CreatePostInHome/CreatePostInHome';

function HomePageContainer() {
  const theme = useTheme();
  const match = useMediaQuery(theme.breakpoints.up('md'));
  return (
    <OuterContainer>
      <MainContainer>
        <MainContent>
          <CreatePostInHome />
          <PostsClassification />
          <Post />
          <Post />
          <Post />
          <Post />
          <Post />
        </MainContent>
        {match
        && (
        <SideBar>
          <Communities />
          <RedditPremium />
          <PersonalReddit />
          <HomePageFooter />
        </SideBar>
        )}
      </MainContainer>
    </OuterContainer>
  );
}

export default HomePageContainer;
