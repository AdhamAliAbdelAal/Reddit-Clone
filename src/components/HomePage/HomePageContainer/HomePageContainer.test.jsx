import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/system';
import HomePageContainer from './HomePageContainer';
import theme from '../../../styles/theme';
import PostTypeContextProvider from '../../../contexts/PostTypeContext';

describe(HomePageContainer, () => {
  it('HomePageContainer renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <PostTypeContextProvider>
            <ThemeProvider theme={theme}>
              <HomePageContainer />
            </ThemeProvider>
          </PostTypeContextProvider>
          ,
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
