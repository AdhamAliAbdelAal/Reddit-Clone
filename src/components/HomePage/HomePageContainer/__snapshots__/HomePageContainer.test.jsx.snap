// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageContainer HomePageContainer renders correctly 1`] = `
Array [
  <div
    className="MuiBox-root css-101ole"
  >
    <div
      className="css-1yelhn"
    >
      <div
        className="css-3ggxsn"
        width={640}
      >
        <div
          className="css-ysyxwd"
        >
          <div
            className="css-iytt7u"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="avatar"
                className="css-fg9jqr"
                src="https://styles.redditmedia.com/t5_758ciw/styles/profileIcon_snoodd8b11a2-0e4a-4403-a861-a9fa7474b850-headshot.png?width=256&height=256&crop=256:256,smart&s=bc53006491e647452f185afa69775cd6a241598c"
              />
            </a>
          </div>
          <input
            className="css-1txqv5e"
            onClick={[Function]}
            placeholder="Create Post"
            type="text"
          />
          <a
            href="/submit"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorThird MuiIconButton-sizeMedium css-ibn9bp-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="InsertPhotoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-4.86 8.86-3 3.87L9 13.14 6 17h12l-3.86-5.14z"
                />
              </svg>
            </button>
          </a>
          <a
            href="/submit"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorThird MuiIconButton-sizeMedium css-ibn9bp-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-39q5dk-MuiSvgIcon-root"
                data-testid="AttachFileIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                />
              </svg>
            </button>
          </a>
        </div>
        <div
          className="css-1i4l6gc"
        />
      </div>
    </div>
  </div>,
  ",",
]
`;
